### PR TITLE
main.c: code refactoring on the initialization/termination of all the…

### PR DESCRIPTION
**Pull request by andrewhit**
… modules.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -57,6 +57,96 @@ extern bool g_dpvs_pdump;
...
+            }
+        }
+    }
+}
+
+static void modules_term(void)
+{
+    int m, err;
+
+    for (m = MODULE_LAST; m >= MODULE_FIRST; m--) {
```
Comment (ywc689) [inline src/main.c:140]: the initial value for m should be `MODULE_LAST - 1`

File: src/main.c [hunk 2]
```diff
@@ -57,6 +57,96 @@ extern bool g_dpvs_pdump;
...
+        if (dpvs_module_terms[m]) {
+            if ((err = dpvs_module_terms[m]()) != EDPVS_OK) {
```
Comment (ywc689) [inline src/main.c:142]: Some modules don't have cleanup functions. Check whether `dpvs_module_terms[m]` is NULL and call it only when not NULL.

**Review by ywc689 — CHANGES_REQUESTED** [review]

File: src/main.c (hunk 2 above)
Comment (andrewhit) [reply src/main.c:142]: Line 141 checks dpvs_module_terms[m], cleanup func is not-NULL.

File: src/main.c (hunk 1 above)
Comment (andrewhit) [reply src/main.c]: Now the new patch makes "iftraf" as MODULE_LAST.

**Review by ywc689 — APPROVED** [review]

**Pull request merged**